Scripts/RemediationScripts/Remediate-EnableSecurityScanningForContainerRegistry.ps1

```diff
@@ -0,0 +1,716 @@
+
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\EnableSecurityScanningIdentityForContainerRegistry"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+ 
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 3 of 4] Backing up Container Registry(s) details..."
+    Write-Host $([Constants]::SingleDashLine)
+    
+    # Backing up Container Registry(s) details.
+    $backupFile = "$($backupFolderPath)\ContainerRegistryWithoutSecurityScanningEnabled.csv"
+
+    $containerRegistryWithoutSecurityScanningEnabled | Export-CSV -Path $backupFile -NoTypeInformation
+
+    Write-Host "Container Registry(s) details have been backed up to [$($backupFile)]" -ForegroundColor $([Constants]::MessageType.Update)
+
+    if (-not $DryRun)
+    {
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "[Step 4 of 4] Grant access to security scanner identities for image scans on Container Registry(s) in the Subscription..." 
+        Write-Host $([Constants]::SingleDashLine)
+        
+
+        if (-not $Force)
+        {
+            Write-Host "Do you want to grant access to security scanner identities for image scans on Container Registry(s) in the Subscription? " -ForegroundColor $([Constants]::MessageType.Warning)
+            
+            $userInput = Read-Host -Prompt "(Y|N)"
+
+            if($userInput -ne "Y")
+            {
+                Write-Host "Access is not granted to security scanner identities for image scans on Container Registry(s) in the Subscription. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
+                break
+            }
+        }
+        else
+        {
+            Write-Host "'Force' flag is provided. Access will be granted to security scanner identities for image scans on Container Registry(s) in the Subscription without any further prompts." -ForegroundColor $([Constants]::MessageType.Warning)
+        }
+
+        # List for storing remediated Container Registry(s)
+        $containerRegistryRemediated = @()
+
+        # List for storing skipped Container Registry(s)
+        $containerRegistrySkipped = @()
+
+        Write-Host "Creating Reader role for security scanner identities for image scanning on Container Registry(s)." -ForegroundColor $([Constants]::MessageType.Info)
+
+        # Loop through the list of Container Registry(s) which needs to be remediated.
+        $containerRegistryWithoutSecurityScanningEnabled | ForEach-Object {
+            $containerRegistry = $_
+            try
+            {
+                $roleAssignment = New-AzRoleAssignment -Scope $_.ResourceId -ObjectId $ObjectId -RoleDefinitionName "Reader" -ObjectType 'ServicePrincipal'
```

Comment:
New-AzRoleAssignment : A parameter cannot be found that matches parameter name 'ObjectType'.


Plz check in which Az.Account version objectType param is supported and add that in preq.